### ISSUE TITLE
group-X-hover styles apply after press on mobile devices

### ISSUE THREAD

**Issue by jbean96** [tamagui/tamagui#2789]
### Current Behavior

I've found that if I have a component that uses a `$group-{groupName}-hover` style and that component is viewed on a mobile device (done by both using the device simulator on chrome dev tools and my own iPhone on some of our production code) then when the component is pressed the hover style persists even after the press is complete.

Important to note that this does not affect desktop devices, it only seems to be mobile browser agents.

### Expected Behavior

I'd expect that after press and after I remove my cursor from the element, neither `$group-{groupName}-hover` nor `$group-{groupName}-press` styles are applied.

### Tamagui Version

```markdown
1.102.1
```

### Platform (Web, iOS, Android)

```markdown
Web (viewing with a mobile device)
```

### Reproduction

```markdown
https://github.com/jbean96/tamagui/tree/josh/group-hover-bug
```

### System Info

```markdown
System:
    OS: macOS 13.6.7
    CPU: (10) arm64 Apple M1 Pro
    Memory: 3.18 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.18.0 - ~/.nvm/versions/node/v18.18.0/bin/node
    Yarn: 4.1.0 - ~/.nvm/versions/node/v18.18.0/bin/yarn
    npm: 9.8.1 - ~/.nvm/versions/node/v18.18.0/bin/npm
    Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 126.0.6478.127
    Safari: 16.6
  npmPackages:
    @biomejs/biome: ^1.7.3 => 1.7.3 
    @manypkg/cli: ^0.19.1 => 0.19.2 
    @nderscore/tamagui-typescript-plugin: ^0.6.0 => 0.6.0 
    @playwright/test: ^1.40.1 => 1.40.1 
    @types/eslint: ^8 => 8.56.2 
    @typescript-eslint/eslint-plugin: ^6.19.1 => 6.19.1 
    @typescript-eslint/parser: ^6.19.1 => 6.19.1 
    @vercel/og: ^0.5.8 => 0.5.8 
    eslint: ^8.56.0 => 8.56.0 
    eslint-parallel: ^1.2.1 => 1.2.1 
    fs-extra: ^11.2.0 => 11.2.0 
    import-sort-style-module: ^6.0.0 => 6.0.0 
    json5: ^2.2.3 => 2.2.3 
    npm-run-all: ^4.1.5 => 4.1.5 
    p-map: ^5.5.0 => 5.5.0 
    patch-package: ^8.0.0 => 8.0.0 
    prompts: 2.1.0 => 2.1.0 
    turbo: ^1.13.3 => 1.13.3 
    typescript: ^5.5.2 => 5.5.2 
    ultra-runner: ^3.10.5 => 3.10.5 
    update-ts-references: ^3.3.0 => 3.3.0 
    zx: ^7.2.3 => 7.2.3
```

**Comment by jbean96** [comment]
For what it's worth, I tried a workaround where I just did `isPressed` and `isHovered` variants on my styled element and then would set some state based on the events `onPressIn`/`onPressOut`/`onHoverIn`/`onHoverOut` but it appears that didn't work either. My guess is that `onHoverOut` didn't get fired when it was supposed to, which is probably the same issue with the groups.

**Comment by natew** [comment]
let me know if 1.105.2 fixes

**Comment by jbean96** [comment]
> let me know if 1.105.2 fixes

@natew doesn't look like it. Updated the repo to the newest version and still seeing the bug.

**Comment by natew** [comment]
Fixed in 6b07cc6409d1139b2e2c8334f78029e270098adc going out now

**Comment by jbean96** [comment]
Nice! Looks like it. Confirmed on the repro branch. 

Thanks @natew.

Also realized that it's possible it was fixed with `1.105.2`, I don't think I fully cleared out `node_modules` and rebuilt when I tested yesterday. Won't make that mistake again!
